tests/flow/test_cms.py

```
@@ -107,6 +114,30 @@ def test_incrby_query(self):
         #     self.assertEqual([1], self.cmd('cms.query', 'test', 'bar'))
         #     self.assertEqual([0], self.cmd('cms.query', 'test', 'nonexist'))
 
+    def test_batchquery(self):
+        self.assertOk(self.cmd('cms.initbydim', 'k1', '100', '5'))
+        self.assertOk(self.cmd('cms.initbydim', 'k2', '100', '5'))
+        self.assertOk(self.cmd('cms.initbydim', 'k3', '100', '6'))
+        self.assertEqual([5], self.cmd('cms.incrby', 'k1', 'v1', '5'))
+        self.assertEqual([5], self.cmd('cms.incrby', 'k2', 'v2', '5'))
+        self.assertEqual([5], self.cmd('cms.incrby', 'k3', 'v3', '5'))
+        self.assertEqual([5, 5], self.cmd('cms.batchquery', 'KEYS', 'k1', 'k2', 'VALUES', 'v1', 'v2'))
+        # k1, k3 have diff depth, error
+        self.assertRaises(ResponseError, self.cmd, 'cms.batchquery', 'KEYS', 'k1', 'k3', 'VALUES', 'v1', 'v3')
+        # at least one key existed, k1 existed
+        self.assertEqual([5, 0], self.cmd('cms.batchquery', 'KEYS', 'k1', 'k4', 'VALUES', 'v1', 'v4'))
```

This should return an error as well.